azurerm/internal/services/vmware/vmware_cluster_resource_test.go

```diff
@@ -0,0 +1,138 @@
+package vmware_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type VmwareClusterResource struct {
+}
+
+func TestAccVmwareCluster_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_vmware_cluster", "test")
+	r := VmwareClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("cluster_id").Exists(),
+				check.That(data.ResourceName).Key("hosts.#").Exists(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccVmwareCluster_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_vmware_cluster", "test")
+	r := VmwareClusterResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
```

Comment:
we are missing the complete test?

Reply:
Since all fields are `Required`, `complete` is the same as the `basic`?

Reply:
ah yep! good shout